### PR TITLE
Part.getDuration takes autoNext.overlap into account.

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
_**Note:** Depends on https://github.com/tv2/sofie-server/pull/11._

### CODE REVIEW

File: src/app/core/models/part.ts [hunk 1]
```diff
@@ -24,6 +26,7 @@ export class Part {
   public expectedDuration?: number
   public executedAt: number
   public playedDuration: number
+  public autoNext: AutoNext | undefined
```
Comment (LindvedKrvang) [inline src/app/core/models/part.ts:29]: Any reason for not writing it as `public autoNext?: AutoNext`?

File: src/app/core/models/part.spec.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+                            const result: number = testee.getDuration()
+
+                            expect(result).toBe(DEFAULT_PART_DURATION_IN_MS)
+                        })
+                    })
+
+                    describe('default part duration is shorter than the time since part was taken', () => {
+                        it('returns the time since part was taken', () => {
+                            const playedDuration: number = 5000
```
Comment (LindvedKrvang) [inline src/app/core/models/part.spec.ts:104]: Maybe do: `const playedDuration: number = DEFAULT_PART_DURATION_IN_MS + 5000`?
Then if DEFAULT_PART_DURATION_IN_MS has its value updated in the future to something higher than 5000, then this test wont break.

**Review by LindvedKrvang — APPROVED** [review]

**Review by RasmusAlbrektsen — APPROVED** [review]

File: src/app/core/models/part.ts (hunk 1 above)
Comment (KvelaGorrrrnio) [reply src/app/core/models/part.ts:29]: No. Is changed now.

File: src/app/core/models/part.spec.ts (hunk 2 above)
Comment (KvelaGorrrrnio) [reply src/app/core/models/part.spec.ts]: Good observation. Is changed now.

**Pull request merged**